src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1558,360 @@ function writeManifestFile(
 
   return store.set('manifest', installedInfo)
 }
+
+export const downloadPatcher = async () => {
+  if (!existsSync(ipdtPatcher)) {
+    try {
+      const { downloadIPDTForOS } = await import('@hyperplay/patcher')
+      await downloadIPDTForOS(toolsPath)
+
+      const version = await getIpdtPatcherVersion()
+      const versionFile = path.join(toolsPath, 'ipdt_version.txt')
+
+      logInfo(
+        `IPDT patcher ${version} downloaded successfully`,
+        LogPrefix.HyperPlay
+      )
+      await writeFile(versionFile, version)
+
+      if (!isWindows) {
+        await chmod(ipdtPatcher, 0o755)
+      }
+    } catch (error) {
+      captureException(error, {
+        extra: {
+          method: 'downloadPatcher'
+        }
+      })
+
+      logError(`Error downloading IPDT: ${error}`, LogPrefix.HyperPlay)
+      throw new Error('Error downloading IPDT')
+    }
+  }
+}
+
+const getIpdtPatcherVersion = async () => {
+  const { stdout } = await spawnAsync(ipdtPatcher, ['-version'])
+  return `${stdout}`.split(' ')[2]
+}
+
+export async function downloadGameIpdtManifest(
+  appName: string,
+  version: string
+) {
+  try {
+    const {
+      install: { channelName, platform },
+      is_installed
+    } = getGameInfo(appName)
+    if (!channelName || !platform || !is_installed) return false
+
+    // download only if the manifest file is not already downloaded and its the same version
+    const manifestName = `${appName}-${platform}-${version}.json`
+    const manifestPath = path.join(ipdtManifestsPath, manifestName)
+    if (existsSync(manifestPath)) return false
+
+    const releaseId = generateID(appName, version)
+    const manifestUrl = await getHyperPlayReleaseManifest(releaseId, platform)
+    if (!manifestUrl) return false
+
+    // download and save the manifest file as a json file in manifest folder
+    logInfo(
+      `Downloading manifest for ${appName} from ${manifestUrl} to ${manifestPath}`,
+      LogPrefix.HyperPlay
+    )
+    await downloadFile(
+      manifestUrl,
+      ipdtManifestsPath,
+      manifestName,
+      createAbortController(appName)
+    )
+      .then(async () => {
+        // remove the empty line at the end of the json file to avoid EOF issues
+        const manifestContent = await readFile(manifestPath, 'utf8')
+        const manifestContentTrimmed = manifestContent.trim()
+        await writeFile(manifestPath, manifestContentTrimmed)
+      })
+      .catch((error) => {
+        throw new Error(`Error downloading manifest for ${appName}: ${error}`)
+      })
```

Comment:
this is already in a try-catch with the same error logging, why are we using a then.catch?

Reply:
Yeah, in my tests there was a race conditions that were fixed when I added this extra then/catch.

Reply:
can you give it another try? the code should be he same the await syntax

Reply:
solved